bcda/cclf/cclf.go

```diff
@@ -275,7 +276,23 @@ func importCCLF(ctx context.Context, fileMetadata *cclfFileMetadata, importer im
 			continue
 		}
 
-		err = importer.do(ctx, txn, cclfFile.ID, b)
+		const (
+			mbiStart, mbiEnd   = 0, 11
+			hicnStart, hicnEnd = 11, 22
+		)
+		cclfBeneficiary := models.CCLFBeneficiary{
+			FileID: cclfFile.ID,
+			MBI:    string(bytes.TrimSpace(b[mbiStart:mbiEnd])),
+			HICN:   string(bytes.TrimSpace(b[hicnStart:hicnEnd])),
+		}
+		// Filtering for duplicate benes in CCLF file
+		if _, ok := importedMBI[cclfBeneficiary.MBI]; ok {
```

Comment:
What kind of performance hit are we taking here?
I'm wondering if just ingesting the duplicate MBIs is simpler/faster than checking every single MBI to see if we've already ingested it.  Remember, we are protecting against sending claims data for duplicate beneficiaries using the work in [this PR](https://github.com/CMSgov/bcda-app/pull/506) and [this PR](https://github.com/CMSgov/bcda-app/pull/521).
Bottom line - is the work and potential performance hit from the work in this PR worth it?

Reply:
So the filtering that we're doing here is going to be in memory, so I'd wager that we won't see any performance difference. I don't think that'll we'll see any appreciable gains on the ingest performance, but don't think we'll come out slower.

I think you hit on the point that we're looking to improve. If we can have a guarantee that we do not have any dupes in the CCLF file, we can leverage simpler queries that can better use the index ( we can use the file_id index). As it is now, we're firing off two queries to get around the dupes, and our final query is leveraging CCLFBeneficiary#id which may not be as efficient since it's aunique column.

Reply:
> So the filtering that we're doing here is going to be in memory, so I'd wager that we won't see any performance difference. I don't think that'll we'll see any appreciable gains on the ingest performance, but don't think we'll come out slower.
> 
> I think you hit on the point that we're looking to improve. If we can have a guarantee that we do not have any dupes in the CCLF file, we can leverage simpler queries that can better use the index ( we can use the file_id index). As it is now, we're firing off two queries to get around the dupes, and our final query is leveraging CCLFBeneficiary#id which may not be as efficient since it's aunique column.

Works for me!